### DISCUSSION TITLE
Drop git hooks in favor of CI-checks

### DISCUSSION THREAD

**Discussion by mark-omarov** [trunklabs/gh-contribution-mate#48] [Ideas]
I think we should drop all git hooks and only use CI-checks. We already have [Semanic PRs](https://github.com/Ezard/semantic-prs) to validate the title of the PR, and we can run lint and format checks on the CI.

These checks take time to run. And while format and lint at least make some sense (although, very little considering it's auto-formatted and linted during development), the conventional commits check makes no sense at all. At the end of the day, all commits will be squashed together and called by the PR title. To preserve the conventional commits history on the main branch we only need to make sure those PRs are merged with the correct commit message (aka PR title).

We can keep using [velociraptor](https://velociraptor.run/) - it has a little bit cleaner task declaration style. Just drop hooks and corresponding scripts, and replace them with a format and lint CI checks.

**Comment by alexey-marinkevich** [comment]
> At the end of the day, all commits will be squashed together and called by the PR title.

I agree on this statement, we indeed don't need pre-commit hooks for now in our situation

**Reply by mark-omarov** [reply]
Here's #49 to track it.
